cmd/kubeapps-apis/plugins/resources/v1alpha1/namespaces_filtering.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+package main
+
+import (
+	"context"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	authorizationapi "k8s.io/api/authorization/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	log "k8s.io/klog/v2"
+	"math"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+type checkNSJob struct {
+	ns corev1.Namespace
+}
+
+type checkNSResult struct {
+	checkNSJob
+	allowed bool
+	Error   error
+}
+
+func (s *Server) MaxWorkers() int {
+	return int(s.clientQPS)
+}
+
+// GetAccessibleNamespaces return the list of namespaces that the user has permission to access
+func (s *Server) GetAccessibleNamespaces(ctx context.Context, cluster string, trustedNamespaces []corev1.Namespace) ([]corev1.Namespace, error) {
+	var namespaceList []corev1.Namespace
+
+	if len(trustedNamespaces) > 0 {
+		namespaceList = append(namespaceList, trustedNamespaces...)
+	} else {
+
+		typedClient, _, err := s.clientGetter(ctx, cluster)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "unable to get the k8s client: '%v'", err)
+		}
+
+		// Try to list namespaces with the user token, for backward compatibility
+		namespaces, err := typedClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			if k8sErrors.IsForbidden(err) {
+				backgroundCtx := context.Background()
```

Comment:
Also, here we're not being consistent with previous uses and are using `context.Background()`, and below we resort to `context.TODO()` again. Also, I guess you could get rid of `backgroundCtx` variable as it's only used in one single place, so it could be inlined.